File: ruby/conintl.rb
# ruby conintl.rb
# http://localhost:4567/?n=10
require 'webrick'

ONES = %w[cero uno dos tres cuatro cinco seis siete ocho nueve diez
          once doce trece catorce quince dieciséis diecisiete dieciocho diecinueve]
TENS = ['', '', 'veinte', 'treinta', 'cuarenta', 'cincuenta',
        'sesenta', 'setenta', 'ochenta', 'noventa']
HUNDREDS = ['', 'ciento', 'doscientos', 'trescientos', 'cuatrocientos',
            'quinientos', 'seiscientos', 'setecientos', 'ochocientos', 'novecientos']

def to_words_es(n)
  n = n.to_i
  return 'cero'   if n == 0
  return ONES[n]  if n < 20
  if n < 100
    return TENS[n / 10] if n % 10 == 0
    return "#{TENS[n / 10]} y #{ONES[n % 10]}"
  end
  if n < 1000
    return 'cien' if n == 100
    rest = to_words_es(n % 100)
    return rest.empty? ? HUNDREDS[n / 100] : "#{HUNDREDS[n / 100]} #{rest}"
  end
  n.to_s
end

server = WEBrick::HTTPServer.new(Port: 4567, Logger: WEBrick::Log.new('/dev/null'), AccessLog: [])

server.mount_proc '/' do |req, res|
  n = (req.query['n'] || '10').to_i
  res.body = to_words_es(n)
end

trap('INT') { server.shutdown }
server.start

File: golang/conintl.go
// go run conintl.go
// http://localhost:8080/?n=10
package main

import (
	"fmt"
	"net/http"
	"strconv"
)

var ones = []string{
	"", "uno", "dos", "tres", "cuatro", "cinco",
	"seis", "siete", "ocho", "nueve", "diez",
	"once", "doce", "trece", "catorce", "quince",
	"dieciséis", "diecisiete", "dieciocho", "diecinueve",
}
var tensWords = []string{
	"", "", "veinte", "treinta", "cuarenta", "cincuenta",
	"sesenta", "setenta", "ochenta", "noventa",
}
var hundreds = []string{
	"", "ciento", "doscientos", "trescientos", "cuatrocientos",
	"quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos",
}

func toWordsES(n int) string {
	switch {
	case n == 0:
		return "cero"
	case n < 20:
		return ones[n]
	case n < 100:
		if n%10 == 0 {
			return tensWords[n/10]
		}
		return tensWords[n/10] + " y " + ones[n%10]
	case n < 1000:
		if n == 100 {
			return "cien"
		}
		rest := toWordsES(n % 100)
		if rest == "" {
			return hundreds[n/100]
		}
		return hundreds[n/100] + " " + rest
	default:
		return strconv.Itoa(n)
	}
}

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		n, _ := strconv.Atoi(r.URL.Query().Get("n"))
		fmt.Fprint(w, toWordsES(n))
	})
	fmt.Println("http://localhost:8080/?n=10")
	http.ListenAndServe(":8080", nil)
}
